pkg/api-server/resource_endpoints.go

```diff
@@ -379,25 +382,27 @@ func (r *resourceEndpoints) createResource(
 func (r *resourceEndpoints) updateResource(
 	ctx context.Context,
 	currentRes model.Resource,
-	newSpec model.ResourceSpec,
+	newResRest rest.Resource,
 	response *restful.Response,
-	labels map[string]string,
 ) {
 	if err := r.resourceAccess.ValidateUpdate(
 		ctx,
 		model.ResourceKey{Mesh: currentRes.GetMeta().GetMesh(), Name: currentRes.GetMeta().GetName()},
 		currentRes.GetSpec(),
-		newSpec,
+		newResRest.GetSpec(),
 		r.descriptor,
 		user.FromCtx(ctx),
 	); err != nil {
 		rest_errors.HandleError(ctx, response, err, "Access Denied")
 		return
 	}
 
-	_ = currentRes.SetSpec(newSpec)
+	_ = currentRes.SetSpec(newResRest.GetSpec())
+	if r.descriptor.HasStatus { // todo(jakubdyszkiewicz) should we always override this?
```

Comment:
All resources on Universal already have `status`. Is it going to simplify the code if we generate `Status` field for all resources and drop the `HasStatus` logic altogether?

As long as we're using a pointer and `omitempty`, the `Status` field won't be visible during marshaling. Also, it'll be always safe to call `SetStatus`.

Reply:
> All resources on Universal already have status

What do you mean?

I was debating whether to implement noop SetStatus instead of if on HasStatus, but my conclusion was that it's less error prone.

Reply:
> > All resources on Universal already have status
> 
> What do you mean?

Just the fact that PostgreSQL transaction creates a `status` column. 

> I was debating whether to implement noop SetStatus instead of if on HasStatus, but my conclusion was that it's less error prone.

Ok if you've already explored the idea then I'm fine with it